### DISCUSSION TITLE
How to set error text for number?

### DISCUSSION THREAD

**Discussion by aspirisen** [colinhacks/zod#618] [Q&A]
I have simple schema where field is just `number`
```
const schema = z.object({
  age: z.number()
});

schema.safeParse({ age: "" });
```

However when i do parsing i see the error `Expected number, received string` what is correct. The question is how can i change the error text?

I thought smth like that, but it doesn't work
```
const schema = z.object({
  age: z.number({ message: 'My number parser error message' })
});
```

Example: https://codesandbox.io/s/zod-error-text-p0w3f?file=/src/App.tsx

UPD: There is pr https://github.com/colinhacks/zod/pull/543

**Comment by scotttrinh** [comment]
@aspirisen 

There is an [error handling document](https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md) that can give you some ideas. If you want to customize it across the board, you can use an "error map" as described [here](https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md#customizing-errors-with-zoderrormap)

**Reply by aspirisen** [reply]
Cool, thanks. However, just curios, why i can't set the message like that `z.number({ message: 'My number parser error message' })`.

**Reply by scotttrinh** [reply]
I believe we're considering a per-schema API for setting error messages, but it hasn't been implemented yet.

**Reply by YimingIsCOLD** [reply]
There is a PR #543 on setting error message on primitives API. Do keep a lookout on it.
